mirdata/billboard.py

```diff
@@ -0,0 +1,453 @@
+    checksum="a7b1fa6a7e454bf73ced7c29207aa597",
+    destination_dir=None,
+)
+
+CHORDS_MIREX13 = download_utils.RemoteFileMetadata(
+    filename="billboard-2.0.1-mirex.tar.xz",
+    url="https://www.dropbox.com/s/fg8lvy79o7etiyc/billboard-2.0.1-mirex.tar.xz?dl=1",
+    checksum="97e5754699f3b45aa5cc70d8a7611c54",
+    destination_dir=None,
+)
+
+
+def _load_metadata(data_home):
+
+    metadata_path = os.path.join(data_home, "billboard-2.0-index.csv")
+
+    if not os.path.exists(metadata_path):
+        logging.info("Metadata file {} not found.".format(metadata_path))
+        return None
+
+    with open(metadata_path, "r") as fhandle:
+        reader = csv.reader(fhandle, delimiter=",")
+        next(reader, None)
+        raw_data = []
+        for line in reader:
+            if line != []:
+                raw_data.append(line)
+
+    metadata_index = {}
+    for line in raw_data:
+        track_id = line[0]
+        metadata_index[track_id] = {
+            "chart_date": line[1],
+            "target_rank": int(line[2]) if line[2] else None,
+            "actual_rank": int(line[3]) if line[3] else None,
+            "title": line[4],
+            "artist": line[5],
+            "peak_rank": int(line[6]) if line[6] else None,
+            "weeks_on_chart": int(line[7]) if line[7] else None,
+        }
+
+    metadata_index["data_home"] = data_home
+
+    return metadata_index
+
+
+DATA = utils.LargeData("billboard_index.json", _load_metadata)
+
+
+class Track(object):
+    """SALAMI Track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        track_id (str): track id
+        audio_path (str): audio path of the track
+        chart date: release date of the track
+        target rank: 
+        actual rank: 
+        title: title of the track
+        artist: artist name
+        peak rank: 
+        weeks on chart: 
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError("{} is not a valid track ID in Salami".format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+
+        metadata = DATA.metadata(data_home)
+        if metadata is not None and track_id in metadata.keys():
+            self._track_metadata = metadata[track_id]
+        else:
+            # annotations with missing metadata
+            self._track_metadata = {
+                "chart_date": None,
+                "target_rank": None,
+                "actual_rank": None,
+                "title": None,
+                "artist": None,
+                "peak_rank": None,
+                "weeks_on_chart": None,
```

Comment:
do you know if there are tracks with missing metadata in Billboard? maybe this is not necessary